sys/kern/uipc_mbuf.c

```diff
@@ -488,8 +498,13 @@ m_dup_pkthdr(struct mbuf *to, const struct mbuf *from, int how)
 #endif
 	to->m_flags = (from->m_flags & M_COPYFLAGS) |
 	    (to->m_flags & (M_EXT | M_EXTPG));
-	if ((to->m_flags & M_EXT) == 0)
-		to->m_data = cheri_kern_setbounds(to->m_pktdat, MHLEN);
+	if ((to->m_flags & M_EXT) == 0) {
```

Comment:
I think it's worth a dedicated commit for all the `m_pktdat` cases, or at least an explicit note in the commit log that we are depending on subobject bounds from the compiler to set exact bounds on `m_pktdat` when setting `m_data` and maybe that we are just ignoring bounds for purecap without subobject?

Reply:
Fixed.